airflow/models/dagrun.py

```diff
@@ -125,6 +131,12 @@ class DagRun(Base, LoggingMixin):
             mssql_where=text("state='queued'"),
             sqlite_where=text("state='queued'"),
         ),
+        ForeignKeyConstraint(
+            [log_filename_id],
+            ["log_filename.id"],
+            name="task_instance_log_filename_id_fkey",
+            ondelete="RESTRICT",
+        ),
```

Comment:
`RESTRICT` means a `LogFilename` row cannot be deleted unless no `DagRun` is referencing it.